packages/gatsby/src/utils/merge-gatsby-config.ts

```diff
@@ -1,30 +1,16 @@
-const _ = require(`lodash`)
-/**
- * Defines how a theme object is merged with the user's config
- */
-module.exports = (a, b) => {
-  // a and b are gatsby configs, If they have keys, that means there are values to merge
-  const allGatsbyConfigKeysWithAValue = _.uniq(
-    Object.keys(a).concat(Object.keys(b))
-  )
-
-  // reduce the array of mergable keys into a single gatsby config object
-  const mergedConfig = allGatsbyConfigKeysWithAValue.reduce(
-    (config, gatsbyConfigKey) => {
-      // choose a merge function for the config key if there's one defined,
-      // otherwise use the default value merge function
-      const mergeFn = howToMerge[gatsbyConfigKey] || howToMerge.byDefault
-      return {
-        ...config,
-        [gatsbyConfigKey]: mergeFn(a[gatsbyConfigKey], b[gatsbyConfigKey]),
-      }
-    },
-    {}
-  )
+import _ from "lodash"
+import { GatsbyConfig } from "../../index"
```

Comment:
Let's use https://github.com/gatsbyjs/gatsby/blob/63c9048b0f09721009e6c9c6963d706c602583a7/packages/gatsby/src/redux/types.ts#L38-L59 instead of https://github.com/gatsbyjs/gatsby/blob/63c9048b0f09721009e6c9c6963d706c602583a7/packages/gatsby/index.d.ts#L162-L195

We use `IGatsbyConfig` in other parts that were already converted to TS - we potentially can expand our current `IGatsbyConfig` interface using types we currently export from package. We will later on likely change or remove top level `index.d.ts` to re-export types from codebase, so this swap would save us some future work

Reply:
I agree with it. There's a difference between GatsbyConfig and IGatsbyConfig, so I added `mapping` property in IGatsbyConfig